rules/linux/persistence_rc_script_creation.toml

```diff
@@ -17,10 +34,67 @@ boot. Adversaries may alter rc.local to execute malicious code at start-up, and
 system. 
 """
 from = "now-9m"
-index = ["logs-endpoint.events.*", "auditbeat-*", "endgame-*"]
+index = ["logs-endpoint.events.*", "endgame-*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "Potential Persistence Through Run Control Detected"
+note = """## Triage and analysis
+
+### Investigating Potential Persistence Through Run Control Detected
+
+The `rc.local` file is used for executing custom commands or scripts during system startup on Linux systems. `rc.local` has been deprecated in favor of the use of `systemd services`, and more recent Unix distributions no longer leverage this method of on-boot script execution. 
+
+There might still be users that use `rc.local` in a benign matter, so investigation to see whether the file is malicious is vital. 
+
+Detection alerts from this rule indicate the creation of a new `/etc/rc.local` file. 
+
+> **Note**:
+> This investigation guide uses the [Osquery Markdown Plugin](https://www.elastic.co/guide/en/security/master/invest-guide-run-osquery.html) introduced in Elastic Stack version 8.5.0. Older Elastic Stack versions will display unrendered Markdown in this guide.
+
+#### Possible Investigation Steps
+
+- Identify the user account that performed the action and whether it should perform this kind of action.
+- Investigate the file that was created or modified.
+  - $osquery_0
+- Investigate the script execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence and whether they are located in expected locations.
+  - $osquery_1
+- Investigate whether the `/lib/systemd/system/rc-local.service` and `/run/systemd/generator/multi-user.target.wants/rc-local.service` have been created through the `systemd-rc-local-generator` located at `/usr/lib/systemd/system-generators/systemd-rc-local-generator`.
+  - $osquery_2
+  - In case the file is not present here, `sudo systemctl status rc-local` can be executed to find the location of the rc-local unit file.
+  - If `rc-local.service` is found, manual investigation is required to check for the rc script execution. Systemd will generate syslogs in case of the execution of the rc-local service. `sudo cat /var/log/syslog | grep "rc-local.service|/etc/rc.local Compatibility"` can be executed to check for the execution of the service.
+    - If logging is found, analyze it, and chances are that the contents of the `rc.local` file have been executed. In case several syslog log files are available, use a wildcard to search through all of the available logs.
```

Comment:
```suggestion
    - If logs are found, it's likely that the contents of the `rc.local` file have been executed. Analyze the logs. In case several syslog log files are available, use a wildcard to search through all of the available logs.
```